src/audio/component.c

```diff
@@ -29,20 +29,66 @@ DECLARE_SOF_UUID("component", comp_uuid, 0x7c42ce8b, 0x0108, 0x43d0,
 
 DECLARE_TR_CTX(comp_tr, SOF_UUID(comp_uuid), LOG_LEVEL_INFO);
 
-static const struct comp_driver *get_drv(uint32_t type)
+static const struct comp_driver *get_drv(struct sof_ipc_comp *comp)
 {
 	struct comp_driver_list *drivers = comp_drivers_get();
 	struct list_item *clist;
 	const struct comp_driver *drv = NULL;
 	struct comp_driver_info *info;
+	struct sof_ipc_comp_new_ext *comp_ext;
 	uint32_t flags;
 
 	irq_local_disable(flags);
 
+	/* do we have extended data ? */
+	if (!comp->ext_data_length)
+		goto comp_type_match;
+
+	/* validate the hdr.size */
+	if (comp->hdr.size < comp->ext_data_length) {
+		tr_err(&comp_tr, "Invalid size, hdr.size:0x%x, ext_data_length:0x%x\n",
```

Comment:
Assuming I got this right, replace with:
```C
// Basic sanity check of total length and extended data length.
// In this generic code we don't know which specific comp and how much it adds
if (comp->hdr.size < sizeof(sof_ipc_comp) + comp->ext_data_length) {

```
-  stricter so catches more errors
- makes it clearer that `hdr.size` is actually `hdr.total_size` without changing its name.

Reply:
ack - with C style comments.

Reply:
@marc-hb, this is not included in your fixup commits?